src/clr-angular/forms/select/select-menu.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component, ElementRef, HostBinding, Inject, Injector, Optional} from "@angular/core";
+import {AbstractPopover} from "../../popover/common/abstract-popover";
+import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
+import {Point} from "../../popover/common/popover";
+import {Subscription} from "rxjs/Subscription";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+@Component({
+    selector: "clr-select-menu",
+    templateUrl: "select-menu.html",
+    host: {"[class.clr-select-menu]": "true"}
+})
+export class ClrSelectMenu extends AbstractPopover {
+    // Have to declare this separately or TS complains that super call in the constructor
+    // should be before the if condition which we do not want in this case
+    private _ifOpenService: IfOpenService;
+
+    constructor(
+        injector: Injector,
+        @Optional() @Inject(POPOVER_HOST_ANCHOR) parentHost: ElementRef,
+        ifOpenService: IfOpenService
+    ) {
+        if (!parentHost) {
+            throw new Error("clr-select-menu should only be used inside of a clr-select");
+        }
+        super(injector, parentHost);
+
+        //Initialize Services
+        this._ifOpenService = ifOpenService;
+
+        //Configure Popover
+        this.initializeSubscriptions();
+        this.configurePopover();
+    }
+
+    @HostBinding("class.open")
+    get open(): boolean {
+        return this._ifOpenService.open;
+    }
+
+    /**
+     * Configure Popover Direction and Close indicators
+     */
+    private configurePopover(): void {
+        this.anchorPoint = Point.BOTTOM_LEFT;
+        this.popoverPoint = Point.LEFT_TOP;
+        this.closeOnOutsideClick = true;
+    }
+
+    private initializeSubscriptions(): void {
+        if (this._ifOpenService) {
+            const sub: Subscription = this._ifOpenService.ignoredElementChange.subscribe((el: ElementRef) => {
+                if (el) {
+                    this.ignoredElement = el.nativeElement;
+                }
+                sub.unsubscribe();
```

Comment:
Now that I think about this, I should add the unsubscribe in the onDestroy itself. If by chance `open` doesnt change, this subscription will never unsubscribe.

Reply:
cc: @youdz